Python/specialize.c

```diff
@@ -696,8 +702,40 @@ _Py_Specialize_LoadAttr(PyObject *owner, _Py_CODEUNIT *instr, PyObject *name)
             goto fail;
         }
         case PROPERTY:
-            SPECIALIZATION_FAIL(LOAD_ATTR, SPEC_FAIL_ATTR_PROPERTY);
-            goto fail;
+        {
+            _PyLoadMethodCache *lm_cache = (_PyLoadMethodCache *)(instr + 1);
+            assert(Py_TYPE(descr) == &PyProperty_Type);
+            PyObject *fget = ((_PyPropertyObject *)descr)->prop_get;
+            if (fget == NULL || Py_TYPE(fget) != &PyFunction_Type) {
+                SPECIALIZATION_FAIL(LOAD_ATTR, SPEC_FAIL_EXPECTED_ERROR);
```

Comment:
`fget == NULL` is an error, but `fget` can be any callable.
```
class C:
    x = property(int)
    def __int__(self): return 3

>>> C().x
3
```
So maybe fail with `SPEC_FAIL_ATTR_PROPERTY` if `fget` is not a Python function.

Reply:
Hmm, I wonder how common this form is. Should I add a branch to the `LOAD_ATTR_PROPERTY` opcode? It shouldn't be too difficult.:

```
LOAD_ATTR_PROPERTY:
    if (Py_TYPE(fget) == &PyFunction_Type && tstate->interp->eval_frame == NULL) {
        // Do inline call here
        DISPATCH();
    }
    PyObject *res = PyObject_VectorCall(prop, self);
    // ...
    DISPATCH();
```

Reply:
> I wonder how common this form is

That's what the stats are for :slightly_smiling_face: 
Just add the stats for now, then we can see if it is worth adding another case.

In general, there should be few branches, ideally none, after the `DEOPT_IF`s in a specialized instruction.